### DISCUSSION TITLE
How does "Used by" work?

### DISCUSSION THREAD

**Discussion by Bibo-Joshi** [community/community#21635] [New to GitHub]
<p>Hi!</p>
<p>There are already some topics on problems with the “Used by” feature, about dependants not being listed or GitHub failing to identify the packages that the repository provides.</p>
<p>The latter case is the trigger for this topic. More specifically, GitHub doesn’t infere that the repository <a href="https://github.com/python-telegram-bot/python-telegram-bot" rel="noopener nofollow ugc">here</a> provides the packages <code>python-telegram-bot</code> and <code>python-telegram-bot-raw</code> on <a href="http://pypi.org" rel="noopener nofollow ugc">pypi.org</a>. Admittedly, it’s probably not easy, as in <code>setup.py</code> we have something like <code>name=f'python-telegram-bot{raw_ext}'</code>.</p>
<p>Still, I’d of course like for the dependents to be listed. So my question is basically two-fold:</p>
<ol>
<li>How can I achievie this for my specific use case?</li>
<li>Is there any documentation on how exactly GitHub checks if a repository contains a package? <a href="https://docs.github.com/en/code-security/supply-chain-security/about-the-dependency-graph#supported-package-ecosystems" rel="noopener nofollow ugc">Here</a> I found which files are checked for dependencies, but not for dependents . If there is not, can it be added somewhere? I’ll happily PR to the docs, but I would need the info :slight_smile:
</li>
</ol>
<p>Bonus question: If GitHub can not automatically infere the provided package/s, can we have an option to manually set those? That could e.g. be verified by checking that the GitHub repo referenced on the project page of the corresponding package manager.</p>
<p>I’d be happy for hints!</p>

**Comment by ettaboyle** [comment]
<p>Hi <a class="mention" href="/bibo-joshi">@Bibo-Joshi</a> :wave: We saw that this was your first post and wanted to welcome you to the GitHub Community &amp; the forum :tada:</p>
<p>A member of the Community team saw your post and they are reaching out to the product manager to share a link to your post and to request some feedback. If there is anything else you want to share please keep commenting in the thread.</p>
<p>Again, welcome! We’re happy you’re here :blush:</p>

**Comment by Bibo-Joshi** [comment]
<p>Hi <a class="mention" href="/ettaboyle">@ettaboyle</a> , thanks for your kind reply :slight_smile: Happy to hear that my question is being forwarded!</p>

**Comment by ernest-phillips** [comment, marked as answer]
<p>Hi <a class="mention" href="/bibo-joshi">@Bibo-Joshi</a> , thanks for your patience here.</p>
<p>I reassigned the <code>python-telegram-bot</code> and <code>python-telegram-bot-raw</code> packages to your repository.<br>
I suspect that your assumptions about the setup.py file for the name may have made it difficult for our algorithm to find the proper package association.</p>
<p>There currently is no option to manually set these package associations.</p>

**Comment by Bibo-Joshi** [comment]
<p>Thank you very much, <a class="mention" href="/ernest-phillips">@ernest-phillips</a> !</p>

**Comment by Bibo-Joshi** [comment]
<p>Actually, I have a follow-up question (please tell, if I should instead open a new topic): IISC packages that are not shipped via a package manager but can be installed from a github repo directly are not tracked. E.g. our repo <a href="https://github.com/python-telegram-bot/ptbcontrib" rel="noopener nofollow ugc">ptbcontrib</a> would be listed in a <code>requirements.txt</code> as <code>git+https://github.com/python-telegram-bot/ptbcontrib.git</code>. Would it maybe be an idea to track that aswell? From my naive point of few that should be pretty easy to parse from requirements files … Then again, I don’t know how common that is for languages other than python …</p>

**Comment by ernest-phillips** [comment]
<p>Those types of packages are not tracked and would be overlooked for manifest detection.</p>
<p>If you like, you can submit a request for this functionality through our official <a href="https://support.github.com/contact/feedback">product feedback form</a> so that our product team can track your request.</p>
<p>You can also keep an eye on our <a href="https://github.blog/changelog/">Changelog</a> as well as the <a href="https://github.com/github/roadmap">GitHub public roadmap</a> to see features we are working on.</p>

**Comment by Bibo-Joshi** [comment]
<p>Allright, will do that :slight_smile: Thanks!</p>
